ml-agents/mlagents/trainers/torch/agent_action.py

```diff
@@ -59,3 +61,83 @@ def from_buffer(buff: AgentBuffer) -> "AgentAction":
                 discrete_tensor[..., i] for i in range(discrete_tensor.shape[-1])
             ]
         return AgentAction(continuous, discrete)
+
+    @staticmethod
+    def _padded_time_to_batch(
+        agent_buffer_field: AgentBufferField, dtype: torch.dtype = torch.float32
+    ) -> List[torch.Tensor]:
+        """
+        Pad actions and convert to tensor. Pad the data with 0s where there is no
+        data. 0 is used instead of NaN because NaN is not a valid entry for integer
+        tensors, as used for discrete actions.
+        """
+        action_shape = None
+        for _action in agent_buffer_field:
+            if _action:
```

Comment:
What possible values can _action take? Is this check really needed ?

Reply:
_action could be an empty list if there are no group actions at that step. This check goes through and finds the first non-empty list. Added a comment.